Dockerfile and docker-compose

mauroferra opened the pull request:
Added a Dockerfile to build an image from the latest Git Release of the mining-pool repo, and a docker-compose file to deploy that image along with redis and a Postgres.

Readme updated to include info about docker deployment.

mauroferra committed:
Docker setup initial commit

Added Dockerfile, docker-compose and updated readme

mauroferra committed:
Merge branch 'alephium:master' into docker

mauroferra committed:
Always fetch latest pool release, multi-arch image

mauroferra committed:
Merge branch 'docker' of https://github.com/mauroferra/mining-pool into docker

polarker commented on docker-compose.yaml:
```diff
@@ -0,0 +1,56 @@
+
+services:
+  mining-pool:
+    image: mauroferra/mining-pool:v1.3.0
```
I guess we need to change the prefix `mauroferra`?

mauroferra replied:
It points to an image that I built with the Dockerfile included, which fetches the code from a release tarball, because the official mining-pool image is quite outdated

polarker replied:
We will not include an unofficial docker image in the reference code. Could you remove the image name and ask people to build the docker manually?

mauroferra replied:
ok, i can do that, there's the Dockerfile included just for that. I'll make 2 versions, one that uses the current context (useful for building an image from master with latest code) and one that pulls the zip from the latest release, just like it is now.

Will do in a bit today.

polarker replied:
Take your time, thanks!

mauroferra replied:
Done!

mauroferra committed:
Comment resolution

- Added Dockerfile for local build
- Removed reference to external image
- Added instructions to build from source, latest master code and latest release

JimmyHoffa commented on README.md:
````diff
@@ -92,3 +96,78 @@ npm install
+
+1. From local sources:
+```shell
+docker build -t mining-pool:latest .
````
with the `build:` section inside of compose, you can make this automatic :+1: 

It will name and tag what it builds locally with what you place under `image: name:tag` 

https://docs.docker.com/compose/compose-file/build/

mauroferra replied:
Thanks!

I added the step and adjusted the readme

mauroferra committed:
docker-compose auto build

Added automatic image build step and adjusted readme

Lbqds reviewed (approved):
Sorry, forgot this, just tested it, LGTM 👍

JimmyHoffa committed:
Fixing up config for compose and shrinking docker image

mauroferra committed:
Merge pull request #1 from JimmyHoffa/docker-alpine

Fixing up config for compose and shrinking docker image

mauroferra committed:
Latest changes on Dockerfile-release
